dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/monitor/DatadogRumMonitor.kt

```diff
@@ -65,15 +65,20 @@ internal class DatadogRumMonitor(
             eventData,
             attributes
         )
-        activeViewEvent = event
-        activeViewData = eventData
-        activeViewKey = WeakReference(key)
+
+        synchronized(this) {
+            activeViewEvent = event
```

Comment:
the reason why I wanted to have these volatile is to guarantee they are going to be kept in the main RAM memory and not in the Thread local memory (to avoid cases where if we start from 2 threads they will have their own values). Are you sure the `synchronized` block guarantees that ?